appcoins-ads/src/main/java/com/asf/appcoins/sdk/ads/poa/manager/WalletUtils.java

```diff
@@ -7,48 +7,96 @@
 import android.content.Intent;
 import android.content.pm.PackageManager;
 import android.net.Uri;
+import android.util.ArrayMap;
+import android.util.Log;
 import com.asf.appcoins.sdk.ads.BuildConfig;
 import com.asf.appcoins.sdk.ads.R;
+import java.lang.reflect.Field;
 
-public class WalletUtils{
+public class WalletUtils {
 
-  private static String walletPackageName = BuildConfig.BDS_WALLET_PACKAGE_NAME;
+  private static boolean dialogVisible;
 
-  public static boolean hasWalletInstalled(Context context) {
+  public static String walletPackageName = BuildConfig.BDS_WALLET_PACKAGE_NAME;
+
+  public static Context context;
+
+  public static boolean isDialogVisible() {
+    return dialogVisible;
+  }
+
+  public static void setContext(Context cont) {
+    context = cont;
+  }
+
+  public static boolean hasWalletInstalled() {
     PackageManager packageManager = context.getPackageManager();
 
     try {
       packageManager.getPackageInfo(walletPackageName, 0);
       return true;
     } catch (PackageManager.NameNotFoundException e) {
-      e.printStackTrace();
       return false;
     }
   }
 
-  public static void promptToInstallWallet(Activity activity, Context context, String message,
-      DialogVisibleListener dialogVisibleListener) {
-    showWalletInstallDialog(activity,context,message,dialogVisibleListener);
-  }
+  public static void promptToInstallWallet() {
+    final Activity act;
+    try {
+      act = getActivity();
+    } catch (Exception e) {
+      e.printStackTrace();
+      return;
+    }
 
-  private static void showWalletInstallDialog( final Activity activity,Context context,String message,final DialogVisibleListener dialogVisibleListener) {
+    if (act == null) {
+      Log.d(WalletUtils.class.getName(),"Problem retrieving main Activity");
+      return;
+    }
 
+    dialogVisible = true;
     AlertDialog.Builder builder;
-    builder = new AlertDialog.Builder(context);
+    builder = new AlertDialog.Builder(act);
     builder.setTitle(R.string.wallet_missing);
-    builder.setMessage(message);
+    builder.setMessage(act.getString(R.string.install_wallet_from_iab));
+    Log.d("String name: ", act.getString(R.string.install_wallet_from_iab));
+
     builder.setPositiveButton(R.string.install, new DialogInterface.OnClickListener() {
-      @Override public void onClick(DialogInterface dialogInterface, int i) {
-        activity.startActivity(
+      @Override public void onClick(DialogInterface dialog, int which) {
+        act.startActivity(
             new Intent(Intent.ACTION_VIEW, Uri.parse("market://details?id=" + walletPackageName)));
       }
     });
+
     builder.setNegativeButton(R.string.skip, new DialogInterface.OnClickListener() {
       @Override public void onClick(DialogInterface dialogInterface, int i) {
-        dialogVisibleListener.OnDialogVisibleListener(false);
+        dialogInterface.cancel();
```

Comment:
the method that should be called is dismiss and not cancel. And in this case in both button actions, positive and negative.